common/src/main/java/net/xolt/freecam/mixins/GuiMixin.java

```diff
@@ -34,7 +27,7 @@ private void onGetCameraPlayer(CallbackInfoReturnable<Player> cir) {
     // Don't render pumpkin overlay while Freecam is active
     @Inject(method = "renderTextureOverlay", at = @At("HEAD"), cancellable = true)
     private void onRenderTextureOverlay(GuiGraphics guiGraphics, ResourceLocation shaderLocation, float alpha, CallbackInfo ci) {
-        if (Freecam.isEnabled() && shaderLocation.equals(PUMPKIN_BLUR_LOCATION)) {
+        if (Freecam.isEnabled() && shaderLocation.equals(ResourceLocation.withDefaultNamespace("misc/pumpkinblur"))) {
```

Comment:
Do you think we still need to check it is the pumpkin blur texture?

My thought process is that since the overlay is now data-driven it could be any texture (and be caused by any equipped item), but regardless of what the overlay is it should only apply to the player and not freecam's camera :thinking: 

```suggestion
        if (Freecam.isEnabled()) {
```

Thoughts?

Reply:
Yeah, that probably makes sense

Reply:
Seems the equality check wasn't actually working anyway; I was still seeing the overlay when testing locally... I guess `.equals()` was doing reference-based equality.

Your latest version works correctly for me.